src/components/TimelineNav/TimelineNav.tsx

```diff
@@ -139,23 +138,28 @@ export const TimelineNav = ({
     }
   }, [prevActiveIndex, activeIndex, timelineNavItemRefs]);
 
+  const generatedId = useId(id);
+  const idPrefix = String(generatedId);
+  const generatedAriaLabelledBy = useId(other['aria-labelledby']);
+  const ariaLabelledByPrefix = String(generatedAriaLabelledBy);
+
   /**
    * Autogenerate ids on tabs if not defined.
    */
   useEffect(() => {
     setId(
-      timelineNavItems().map((item) =>
-        item.props.id ? item.props.id : getUID(item),
+      timelineNavItems().map((item, index) =>
+        item.props.id ? item.props.id : `${idPrefix}-${index}`,
       ),
     );
     setAriaLabelledBy(
-      timelineNavItems().map((item) =>
+      timelineNavItems().map((item, index) =>
         item.props['aria-labelledby']
           ? item.props['aria-labelledby']
-          : getUID(item),
+          : `${ariaLabelledByPrefix}-${index}`,
       ),
     );
-  }, [timelineNavItems, getUID]);
+  }, [timelineNavItems, idPrefix, ariaLabelledByPrefix]);
```

Comment:
Calling out diff here, because replacing our usage of `useUIDSeed` is a bit more involved. Note: you cannot call `useId()` inside a for-loop because that breaks the rules of react-hooks.